### ISSUE TITLE
Feature Request: Import links directly from xBrowserSync

### ISSUE THREAD

**Issue by TheCakeIsNaOH** [pirate/ArchiveBox#538]
## Type 

 - [x] General question or discussion
 - [x] Propose a brand new feature
 - [ ] Request modification of existing behavior or design

## What is the problem that your feature request solves

I would like to be able to add the links I have bookmarked in [xBrowserSync](https://www.xbrowsersync.org/) directly into ArchiveBox. 

## Describe the ideal specific solution you'd want, and whether it fits into any broader scope of changes

I would like to be able to:
1. Add links directly from the xBrowserSync json format. This would require a parser to be added.
2. Subscribe to my xBrowserSync "sync" (basically account), and update links as needed. This would require specifying URL and password in the configuration and/or in the CLI. This would also require adding decryption, decompression, and parsing of the json.
3. Filter which folders the bookmarks are added from. There are lots of bookmarks I have that have no need to be in archivebox. This is part where I am not sure if there are more changes that would be required, specifically if filtering inputted links is more global, or if it is just for this. 

The json format is the same for both the backup/export from the extension, and directly from the API. From the web API, it first has to be decrypted (AES-GCM), and then decompressed (lzutf8).

I do currently have a python script for doing this, but I'm not sure what changes would need to be made to integrate with ArchiveBox. I'm new to python, so there probably lots of things that aren't ideal python practice. 
https://github.com/TheCakeIsNaOH/xbs-to-archivebox

I'm definitely willing to work on developing this, I'm just checking in / looking for feedback before starting work.

Useful links:
https://api.xbrowsersync.org/
https://github.com/xbrowsersync/app/issues/105
https://gist.github.com/nero120/e878e40b14655c9526680472376b4f8c
https://github.com/b-01/lzutf8.py

## What hacks or alternative solutions have you tried to solve the problem?

I currently use a python script to download, decompress, and filter my xBrowserSync bookmarks, and it outputs a list of URLs that I then manually add to ArchiveBox. See the previous section for link. I would prefer that this be built into ArchiveBox.

## How badly do you want this new feature?

 - [ ] It's an urgent deal-breaker, I can't live without it
 - [ ] It's important to add it in the near-mid term future
 - [x] It would be nice to have eventually

---

 - [x] I'm willing to contribute dev time / money to fix this issue
 - [x] I like ArchiveBox so far / would recommend it to a friend
 - [ ] I've had a lot of difficulty getting ArchiveBox set up

**Comment by pirate** [comment]
Some more discussion here: https://forum.level1techs.com/t/thecakeisnaohs-devember-xbrowsersync-to-archivebox-and-misc-programming/162649/10
